src/pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -209,6 +209,10 @@ def restore_training_state(self) -> None:
     def restore_precision_plugin_state(self) -> None:
         """Restore the precision plugin state from the pre-loaded checkpoint."""
         prec_plugin = self.trainer.precision_plugin
+
+        if isinstance(prec_plugin, ApexMixedPrecisionPlugin) and self.trainer.state.fn == TrainerFn.FITTING:
+            raise RuntimeError("Resuming training with APEX is currently not supported.")
```

Comment:
This only matters in training. We can still resume via trainer.test(), but optimizer states don't matter there.